src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass_Switch.cs

```diff
@@ -273,6 +290,47 @@ private LocalState VisitPatternSwitchHeader(BoundPatternSwitchStatement node)
             }
         }
 
+        private bool FullyHandlesItsInput(DecisionTree decision)
+        {
+            if (decision == null) return false;
```

Comment:
When there is nothing to do. For example, the WhenNull decision subtree is null when there is no "case null". Rather than doing the test in each caller, we consolidate the test here.

---
In reply to: [70301861](https://github.com/dotnet/roslyn/pull/12416#discussion_r70301861) [](ancestors = 70301861)